### ISSUE TITLE
Discuss to deprecate Prometheus remote write exporter external labels support

### ISSUE THREAD

**Issue by rakyll** [open-telemetry/opentelemetry-collector#3016]
Prometheus remote write exporter (https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/prometheusremotewriteexporter) supports `external_labels` to allow users to attach a set of labels to every outgoing sample. Similar functionality in Prometheus is provided by the Prometheus config by `global.external_labels` and the labels are attached to scraped samples.

To be able to compliant with the Prometheus configuration, we should support allowing users to set external labels in the receiver configuration. Once we have this feature, we might be deprecating the external labels support in the remote write exporter.

See https://github.com/open-telemetry/opentelemetry-collector/issues/2904.

**Comment by rakyll** [comment]
@bogdandrutu @tigrannajaryan Can we get to this issue before the collector is stable?

**Comment by bogdandrutu** [comment]
@rakyll if the incoming data are not from the prometheus receiver this feature is still useful I believe.

**Comment by bogdandrutu** [comment]
I am happy to remove it if nobody uses it.

**Comment by rakyll** [comment]
True. As you say, it could be useful separately from the work needed in the receiver, e.g. OTLP metrics can be labelled by using this feature. But I wonder if we would like to write a processor for that use case instead.

It's your call to keep it or remove it. I don't have any strong opinions.

**Comment by alolita** [comment]
@vishiy @punya Any use cases on your end where this suggested deprecation would have a breaking change?

**Comment by vishiy** [comment]
> @vishiy @punya Any use cases on your end where this suggested deprecation would have a breaking change?

I am good with deprecating this and we have an item to add this to receiver natively. Thanks @alolita

**Comment by punya** [comment]
Thanks for flagging this @alolita. I confirmed that this won't affect us (GCP).

**Comment by rakyll** [comment]
Thanks, everyone. Once #3127 is merged, I will follow up to deprecate the external labels from the remote write exporter.
